src/main/scala/com/ing/wbaa/gargoyle/proxy/data/S3Request.scala

```diff
@@ -21,8 +25,19 @@ object S3Request extends LazyLogging {
       .toList
       .lift(1)
 
-    val accessType = if (httpMethod == HttpMethods.GET) Read else Write
+    val bucketObjectRoot = path.toString
+      .split("/")
+      .toList
+      .lift(2)
+
+    //    val accessType = if (httpMethod == HttpMethods.GET) Read else Write
+    val accessType = httpMethod.value match {
+      case "GET"    => Read
+      case "PUT"    => Write
+      case "POST"   => Write
+      case "DELETE" => Delete
```

Comment:
Maybe best to add a default case _ which defaults to Write?
Also you forgot to remove the code you commented out above